qpc/report/merge.py

```diff
@@ -116,70 +108,17 @@ def _validate_create_json(self, files):
-                    except json_exception_class:
-                        print(_(messages.REPORT_JSON_DIR_FILE_FAILED % file))
-                        continue
-
-                    # validate version type
-                    file_report_version = details_report.get(
-                        REPORT_VERSION_KEY, None)
-                    if not file_report_version:
-                        # warn about old format but continue
-                        print(_(messages.REPORT_MISSING_REPORT_VERSION % file))
-                        file_report_version = DEFAULT_REPORT_VERSION
-
-                    file_report_type = details_report.get(
-                        REPORT_TYPE_KEY, DETAILS_REPORT_TYPE)
-                    if file_report_type != DETAILS_REPORT_TYPE:
-                        # terminate if different from details type
-                        print(_(messages.REPORT_INVALID_REPORT_TYPE %
-                                (file, file_report_type)))
-                        continue
-
-                    # validate sources
-                    sources = details_report.get(SOURCES_KEY, None)
-                    if sources:
-                        has_error = False
-                        for source in sources:
-                            facts = source.get(FACTS_KEY)
-                            server_id = source.get(SERVER_ID_KEY)
-                            if not facts:
-                                print(_(messages.REPORT_JSON_MISSING_ATTR %
-                                        (file, FACTS_KEY)))
-                                has_error = True
-                                break
-                            if not server_id:
-                                print(
-                                    _(messages.REPORT_JSON_MISSING_ATTR %
-                                      (file, SERVER_ID_KEY)))
-                                has_error = True
-                                break
-                            # Add version/type to all sources since merge
-                            source[REPORT_TYPE_KEY] = file_report_type
-                            source[REPORT_VERSION_KEY] = file_report_version
-
-                        if not has_error:
-                            # Source is valid so add it
-                            all_sources += sources
-                            print(_(messages.REPORT_JSON_DIR_FILE_SUCCESS %
-                                    file))
-                    else:
-                        print(_(messages.REPORT_JSON_MISSING_ATTR %
-                                (file, SOURCES_KEY)))
-                        continue
-            else:
-                print(_(messages.FILE_NOT_FOUND % file))
-                sys.exit(1)
+            sources = utils.validate_and_create_json(file)
+            # Source is valid so add it
+            if sources:
+                all_sources += sources
+                print(_(messages.REPORT_JSON_DIR_FILE_SUCCESS %
```

Comment:
Since this prints both here and in in `validate_and_create_json` it creates a duplicate message like the following:
```
Checking details2.json for valid JSON details report.
Success: details2.json is a valid details report.
Success: details2.json is a valid details report.
```

Reply:
👍